packages/desktop-client/src/components/manager/ConfigServer.tsx

```diff
@@ -101,10 +101,23 @@ export function ConfigServer() {
 
     if (selfSignedCertificateLocation) {
       setServerSelfSignedCert(selfSignedCertificateLocation[0]);
-      globalThis.window.Actual.relaunch(); // relaunch to use the certificate
     }
   }
 
+  const isMounted = useRef(false);
+
+  useEffect(() => {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I think this doesn't actually work as intended.

What I think this _should_ do: whenever self signed cert (`serverSelfSignedCert`) changes - the electron server is restarted.

What actually happens:
- electron server is restarted the first time `serverSelfSignedCert` changes
- if the cert changes again (or is removed) - the server does not restart anymore thus is broken again

Reply:
If you move this to a `useCallback` that is called when `setServerSelfSignedCert` is called - that will make it work better IMO.

Reply:
> What actually happens:

> electron server is restarted the first time serverSelfSignedCert changes
> if the cert changes again (or is removed) - the server does not restart anymore thus is broken again

That's not what I'm seeing: 

![test](https://github.com/user-attachments/assets/e8d9f29c-4073-45e0-81e9-502fb56d3ff8)

Reply:
> If the delay is too great, we restart the server before changing the value - that's why I changed it to watch for the value change rather than restating there.

Hmm.. so it all comes down to how long the delay is. And if the delay is sufficiently long - even the `useEffect` solution wouldn't work. Could we `await` for the result of this dispatched action? Or perhaps listen to a callback of some sort?

Reply:
I've passed a callback on the dispatch which will trigger the restart. This feels cleaner than what I had 👍